restore.sh

```diff
@@ -7,10 +7,20 @@ then
     echo "You must pass the path of the backup file to restore"
 fi
 
-echo "=> Restore database from $1"
-set -o pipefail
+SQL=$(gunzip -c "$1")
 DB_NAME=${MYSQL_DATABASE:-${MYSQL_DB}}
-if gunzip --stdout "$1" | mysql -h "$MYSQL_HOST" -P "$MYSQL_PORT" -u "$MYSQL_USER" -p"$MYSQL_PASS" "$DB_NAME"
+if [ -z "${DB_NAME}"]
+then
+    DB_NAME=$(echo $SQL | grep -oE '(Database: (\w*))' | cut -d ' ' -f 2)
```

Comment:
Put $SQL between quotes as the other variables are

Reply:
Sure :)